### ISSUE TITLE
Learning rate scheduler is not working optimally

### ISSUE THREAD

**Issue by alexcwsmith** [LINCellularNeuroscience/VAME#59]
First of all, I have to say I am disappointed that you guys implemented the step size and LR gamma in the new version without asking me to submit a pull request, based on our discussion in https://github.com/LINCellularNeuroscience/VAME/issues/26

Why would it reduce the learning rate 3 epochs after it saved a new best model? See training output below.

It's not like I'm offended by not getting credit or anything like that, but the fact that I suggested something, said I would gladly submit a PR, and you said you'd check it out and let me know, then implemented it in a way that isn't even ideal is annoying to me. The function you probably want instead of the StepLR that is being used right now is:

```
scheduler = ReduceLROnPlateau(optimizer, 'min', factor=cfg['scheduler_gamma'], patience=cfg['scheduler_step_size'], threshold=1e-3, threshold_mode='rel')

```

```
Epoch: 177
Train loss: 35180.000, MSE-Loss: 14238.998, MSE-Future-Loss 20930.243, KL-Loss: 7.326, Kmeans-Loss: 3.433, weight: 1.00
Test loss: 4553.215, MSE-Loss: 4542.580, KL-Loss: 7.351, Kmeans-Loss: 3.284
Saving model!


Epoch: 178
Train loss: 35246.554, MSE-Loss: 14253.390, MSE-Future-Loss 20982.404, KL-Loss: 7.330, Kmeans-Loss: 3.431, weight: 1.00
Test loss: 4602.697, MSE-Loss: 4592.065, KL-Loss: 7.354, Kmeans-Loss: 3.277


Epoch: 179
Train loss: 35200.695, MSE-Loss: 14189.377, MSE-Future-Loss 21000.553, KL-Loss: 7.332, Kmeans-Loss: 3.433, weight: 1.00
Test loss: 4571.499, MSE-Loss: 4560.875, KL-Loss: 7.349, Kmeans-Loss: 3.275


Epoch: 180
Train loss: 35269.871, MSE-Loss: 14263.351, MSE-Future-Loss 20995.757, KL-Loss: 7.329, Kmeans-Loss: 3.433, weight: 1.00
Test loss: 4600.750, MSE-Loss: 4590.125, KL-Loss: 7.343, Kmeans-Loss: 3.282
learning rate update: 1.1881340006899973e-05
```

**Comment by pavolbauer** [comment]
Dear Alex,
First of all, thanks again heaps for all your contributions to VAME!
It seem that in the busy making of VAME 1.0 we have indeed overseen your comment from issue #26.
We have now added your PR and also acknowledged your cotribution in the code:
https://github.com/LINCellularNeuroscience/VAME/blob/b5ca42531b3734f5c965130a01b01541eaf5191b/vame/model/rnn_vae.py#L326
We would be happy to recieve further contributions from you,
Best
Pavol
